UI fixes and code stabilization

lankastersky opened the pull request.

lankastersky committed:
Adding source to Challenge layout. Fixing tests.

lankastersky committed:
Remembering scroll position of selected item in Journal

lankastersky committed:
Showing no challenges message in Journal. Removing unused LaunchScreen

NikAshanin commented on zen/services/StorageService.swift:
```diff
@@ -7,8 +7,12 @@ final class StorageService {
-    init() {
-        userDefaults = UserDefaults(suiteName: StorageService.userDefaultsName)!
+    init(_ userDefaultsName: String) {
+        self.userDefaults = UserDefaults(suiteName: userDefaultsName)!
```
force unwrap, да еще и в init

lankastersky replied:
тут пришлось поплясать немного, чтобы обойти

NikAshanin commented on zen/views/ChallengeViewController.swift:
```diff
@@ -49,10 +52,26 @@ final class ChallengeViewController: UIViewController {
+            return
+        }
+        ChallengeViewController.openUrl(urlString)
+    }
```
что-то вообще перестал следить за отступами, пробелами и тд) между методами - пробел

lankastersky replied:
сорян

NikAshanin commented on zen/views/ChallengeViewController.swift:
```diff
@@ -204,6 +231,18 @@ final class ChallengeViewController: UIViewController {
         scrollView.scrollIndicatorInsets = .zero
     }
+
+    private static func openUrl(_ urlString: String) {
```
а зачем тут статик? почему обычным не сделать?

lankastersky replied:
это больше utility метод. И лучше для перфоманса

NikAshanin commented on zen/views/JournalViewController.swift:
```diff
@@ -22,8 +24,43 @@ final class JournalViewController: UIViewController {
+
+        if let selectedItem = selectedItem {
+            collectionView.scrollToItem(
+                at: IndexPath(row: selectedItem, section: 0),
```
да, джавовский код-стайл это адок конечно)

NikAshanin commented on zen/views/JournalViewController.swift:
```diff
@@ -65,13 +102,12 @@ extension JournalViewController: UICollectionViewDataSource {
 
 extension JournalViewController: UICollectionViewDelegate {
-
+    
```
? удали

lankastersky replied:
готово

NikAshanin commented on zen/views/JournalViewController.swift:
```diff
@@ -65,13 +102,12 @@ extension JournalViewController: UICollectionViewDataSource {
-            "Bad challenge index when creating collection view"
-        )
-
+            "Bad challenge index when creating collection view")
```
вот вообще нечитаемый код тут, из-за этих переносов. давай мб нормально везде сделаем? в стиле

```swift
let a = methodName(firstparam: firstparam,
                   second: second)
```

lankastersky replied:
поправил местами

NikAshanin commented on zen/views/challenge_footer_views/FinishedChallengeFooterView.swift:
```diff
@@ -4,8 +4,12 @@ import Cosmos
+    @IBOutlet private weak var commentsLabel: UILabel!
 
-    @IBOutlet weak var commentsLabel: UILabel!
+    internal var commentsText: String? {
```
internal по дефолту, его ненадо писать

lankastersky replied:
а я думал public по дефолту. Поправил

lankastersky committed:
Code review: formatting fixed, internal word removed

lankastersky committed:
Colors fixed

lankastersky committed:
Deployment target set to iOS 9.
Other languages except Russian are not supported for challenges currently.
Product name changed to Magic Path

NikAshanin commented on zen/services/StorageService.swift:
```diff
@@ -7,8 +7,17 @@ final class StorageService {
-    init() {
-        userDefaults = UserDefaults(suiteName: StorageService.userDefaultsName)!
+    init(_ userDefaultsName: String) {
+        guard let userDefaults = UserDefaults(suiteName: userDefaultsName) else {
```
зачем? есть же специальный init
https://developer.apple.com/swift/blog/?id=17

lankastersky replied:
иначе везде в коде придется использовать optional, что неудобно
